File: sorting/quick_sort.py
from swap import *


def quick_sort(arr):
    if len(arr) <= 1:
        return arr
    else:
        pivot = arr.pop()

    items_greater = []
    items_lower = []

    for item in arr:
        if item > pivot:
            items_greater.append(item)
        else:
            items_lower.append(item)

    return quick_sort(items_lower) + [pivot] + quick_sort(items_greater)


print(quick_sort([34, 5, 7, 90, 1, 23, 4, 6, 100, 23]))

File: sorting/quick_sort.js
const swap =require( "./swap")

const pivot = (arr, start=0, end=arr.length+1) => {
  let pivot = arr[start];
  let sawpIndex = start;
  for (let i = start + 1; i < arr.length; i++) {
    if (pivot > arr[i]) {
      sawpIndex ++;
      swap(arr, sawpIndex, i)
    }
  }
  swap(arr,start,sawpIndex)
  return sawpIndex
}

const quickSort = (arr, left = 0, right = arr.lenght - 1) => {
  if (left < right) {
    let pivotIndex = pivot(arr, left, right)
    //left
    quickSort(arr, left, pivotIndex -1)
    //right
    quickSort(arr, pivotIndex + 1, right)
  }
  return arr;
}

console.log(pivot([4,8,2,1,5,7,6,3]))